app/assets/javascripts/recipes.js

```diff
@@ -11,15 +9,22 @@ $(document).ready(function(){
     });
     if(index > 0 && inputsAreBlank){
       $(grainElement).hide();
-    } else if (index > 0) {
-      $(grainElement).prev(".grain").find("button.add-grain").hide();
-      // Only show button for last non-blank input
     }
-   });
+  });
+
+  var show_last_add_grain_button = function(){
+    $('.grain').filter( function(){ 
+      if($(this).css('display') == "block") {
```

Comment:
Repeated wrapping of $(this) could be cached, see comment below: https://github.com/jpalmieri/brewery/pull/11/files#r24754377